src/bridge/DelayBuffer.sol

```diff
@@ -0,0 +1,177 @@
+    ) internal pure returns (uint256) {
+        if (threshold < prevDelay) {
+            // deplete due to unexpected delay
+            buffer = deplete(start, end, buffer, prevDelay, threshold);
+        } else if (buffer < max) {
+            // replenish depleted buffer
+            buffer = replenish(start, end, buffer, max, replenishRateInBasis);
+        }
+        return buffer;
+    }
+
+    /// @notice Applies full update to buffer data (buffer, sync validity, and prev delay)
+    /// @param self The delay buffer data
+    /// @param blockNumber The update block number
+    function update(BufferData storage self, uint64 blockNumber) internal {
+        // The prevDelay stores the delay of the previous batch which is
+        // used as a starting reference point to calculate an elapsed amount using the current
+        // message as the ending reference point.
+
+        self.bufferBlocks = pendingBufferUpdate(self, blockNumber);
+
+        // store a new starting reference point
+        // any buffer updates will be applied retroactively in the next batch post
+        self.prevBlockNumber = blockNumber;
+        self.prevDelay = uint64(block.number) - blockNumber;
+
+        if (uint64(block.number) - blockNumber <= self.threshold) {
+            updateSyncValidity(self, blockNumber);
+        }
+    }
+
+    /// @dev    The delay buffer can change due to pending depletion / replenishment due to previous delays.
+    ///         This function applies pending buffer changes to calculate buffer updates.
+    /// @notice Calculates the buffer changes up to the requested block number
+    /// @param self The delay buffer data
+    /// @param blockNumber The block number to process the delay up to
+    function pendingBufferUpdate(BufferData storage self, uint64 blockNumber)
+        internal
+        view
+        returns (uint64)
+    {
+        return
+            uint64(
+                bufferUpdate({
+                    start: self.prevBlockNumber,
+                    end: blockNumber,
+                    buffer: self.bufferBlocks,
+                    threshold: self.threshold,
+                    prevDelay: self.prevDelay,
+                    max: self.max,
+                    replenishRateInBasis: self.replenishRateInBasis
+                })
+            );
+    }
+
+    /// @dev    Calculates the margin a sequenced message is below the delay threshold
+    ///         defining a `sync validity` window during which no delay proofs are required.
+    /// @notice Updates the time / block until no delay proofs are required.
+    /// @param blockNumber The block number when the synced message was created.
+    function updateSyncValidity(BufferData storage self, uint64 blockNumber) internal {
+        // saturating at uint64 max handles large threshold settings
+        self.syncExpiry = self.threshold > type(uint64).max - blockNumber
+            ? type(uint64).max
+            : blockNumber + self.threshold;
```

Comment:
```suggestion
        uint256 expiry = blockNumber + self.threshold;
        self.syncExpiry = expiry > type(uint64).max
            ? type(uint64).max
            : uint64(expiry);
```

Reply:
hmmm this suggested version would revert for `self.threshold > type(uint64).max - blockNumber`.

the original is written in a peculiar way, but it permits threshold = type(uint64), or otherwise very large value, to effectively disable the buffer setting a syncExpiry to type(uint64).max

Reply:
but threshold is uint64

Reply:
not sure why it would revert, threshold is uint64 and expiry (`blockNumber + self.threshold`) is uint256

Reply:
I went ahead and made a further simplification which remove the need to update the syncexpiry all together. I think it's a good logical simplification